repocreation/createRepo.go

```diff
@@ -0,0 +1,212 @@
+	}
+
+	dismissalRestrictionsRequest := github.DismissalRestrictionsRequest{
+		Users: &[]string{},
+		Teams: &[]string{dpTeamSlug},
+	}
+	requiredPullRequestReviewsEnforcementRequest := github.PullRequestReviewsEnforcementRequest{
+		DismissalRestrictionsRequest: &dismissalRestrictionsRequest,
+		DismissStaleReviews:          true,
+		RequireCodeOwnerReviews:      true,
+		RequiredApprovingReviewCount: 1,
+	}
+
+	branchRestrictions := github.BranchRestrictionsRequest{
+		Users: []string{},
+		Teams: []string{dpTeamSlug},
+	}
+
+	protectionRequest := github.ProtectionRequest{
+		RequiredStatusChecks:       &requiredStatusChecks,
+		RequiredPullRequestReviews: &requiredPullRequestReviewsEnforcementRequest,
+		EnforceAdmins:              true,
+		Restrictions:               &branchRestrictions,
+	}
+	_, resp, err := client.Repositories.UpdateBranchProtection(ctx, org, repoName, "master", &protectionRequest)
+	if err != nil {
+		log.ErrorCtx(ctx, errors.Wrap(err, "Update branch protection failed for master"), log.Data{"resp": resp})
+	}
+	_, resp, err = client.Repositories.UpdateBranchProtection(ctx, org, repoName, "develop", &protectionRequest)
+	if err != nil {
+		log.ErrorCtx(ctx, errors.Wrap(err, "Update branch protection failed for develop"), log.Data{"resp": resp})
+	}
+	_, resp, err = client.Repositories.RequireSignaturesOnProtectedBranch(ctx, org, repoName, "master")
+	if err != nil {
+		log.ErrorCtx(ctx, errors.Wrap(err, "Adding protection, require signatures failed on branch master"), log.Data{"resp": resp})
+	}
+	_, resp, err = client.Repositories.RequireSignaturesOnProtectedBranch(ctx, org, repoName, "develop")
+	if err != nil {
+		log.ErrorCtx(ctx, errors.Wrap(err, "Adding protection, require signatures failed on branch develop"), log.Data{"resp": resp})
+	}
+	return err
+}
+
+// setDevelopAsDefaultBranch will set the develop branch as the repositories default branch
+func setDevelopAsDefaultBranch(client *github.Client, ctx context.Context, repoName string, repo *github.Repository) error {
+	repo.AutoInit = nil
+	_, resp, err := client.Repositories.Edit(ctx, org, repoName, repo)
+	if err != nil {
+		log.ErrorCtx(ctx, errors.Wrap(err, "Failed to set develop as the default branch"), log.Data{"resp": resp})
+	}
+	return err
+}
+
+// createDevelopBranch will create a new branch named /develop, based on the last ref on master branch
+func createDevelopBranch(client *github.Client, ctx context.Context, repoName string) error {
+	ref, resp, err := client.Git.GetRef(ctx, org, repoName, "heads/master")
+	if err != nil {
+		log.ErrorCtx(ctx, errors.Wrap(err, "Get Reference to master commit failed"), log.Data{"resp": resp})
+		return err
+	}
+	developBranch := "heads/develop"
+	ref.Ref = &developBranch
+
+	_, resp, err = client.Git.CreateRef(ctx, org, repoName, ref)
+	if err != nil {
+		log.ErrorCtx(ctx, errors.Wrap(err, "Create Reference to new develop branch failed"), log.Data{"resp": resp})
+		return err
+	}
+	return nil
```

Comment:
Personal pref but I like to have an "everything was successful" message at the end. Especially if the process has multiple steps that can fail - I just find it useful. I won't block on this though.

Reply:
Ah you will be happy to know that: the entry function `GenerateGithub` (previously know as  `GenerateGithubRepository` has a message "repository has successfully been created please Disable Actions for this repository"